lldb/tools/lldb-dap/Handler/LaunchRequestHandler.cpp

```diff
@@ -21,16 +21,18 @@ using namespace lldb_dap::protocol;
 namespace lldb_dap {
 
 /// Launch request; value of command field is 'launch'.
-Error LaunchRequestHandler::Run(const LaunchRequestArguments &arguments) const {
+void LaunchRequestHandler::Run(
+    const LaunchRequestArguments &arguments,
+    llvm::unique_function<void(llvm::Error)> callback) const {
   // Initialize DAP debugger.
   if (Error err = dap.InitializeDebugger())
-    return err;
+    return callback(std::move(err));
```

Comment:
If I'm reading this correctly, we immediately respond if something goes wrong. Is that compliant with the spec? 

> After the response to configurationDone is sent, the debug adapter may respond to the launch or attach request, and then the debug session has started.

That doesn't seem to limit the response to a successful one?